torchvision/io/video.py

```diff
@@ -260,16 +275,16 @@ def read_video(filename, start_pts=0, end_pts=None, pts_unit="pts"):
         # TODO raise a warning?
         pass
 
-    vframes = [frame.to_rgb().to_ndarray() for frame in video_frames]
-    aframes = [frame.to_ndarray() for frame in audio_frames]
+    vframes_list = [frame.to_rgb().to_ndarray() for frame in video_frames]
+    aframes_list = [frame.to_ndarray() for frame in audio_frames]
 
-    if vframes:
-        vframes = torch.as_tensor(np.stack(vframes))
+    if vframes_list:
+        vframes = torch.as_tensor(np.stack(vframes_list))
     else:
         vframes = torch.empty((0, 1, 1, 3), dtype=torch.uint8)
 
-    if aframes:
-        aframes = np.concatenate(aframes, 1)
+    if aframes_list:
+        aframes = np.concatenate(aframes_list, 1)
         aframes = torch.as_tensor(aframes)
         aframes = _align_audio_frames(aframes, audio_frames, start_pts, end_pts)
```

Comment:
`_align_audio_frames` uses `start_pts` and `end_pts` as indices. Here `end_pts` by default is `float("inf")`, which will not work.

Reply:
At least in the implementation, we have `end > end_ref`, and given that `end_ref` is `inf`, this branch will never hit.
But do you mean that the type annotations will never work here?

Reply:
Correct:

```python
foo = [1, 2, 3]
foo[1:3.0]
```

```
error: Slice index must be an integer or None  [misc]
```

Reply:
Can we just make the type annotations be `Optional[int]`?